extension/goplugin/transaction.go

```diff
@@ -160,7 +160,14 @@ func (t *Transaction) List(ctx context.Context, schema goext.ISchema, filter goe
 	if err := ctx.Err(); err != nil {
 		return nil, 0, ctx.Err()
 	}
-	data, _, err := t.tx.List(context.Background(), t.findRawSchema(schemaID), transaction.Filter(filter), nil, (*pagination.Paginator)(paginator))
+	var o *transaction.ViewOptions
```

Comment:
why this is only used for `List` but not for `Fetch`?

Reply:
i'm not sure why we have such interface for transaction but changing it will break compatibility even more.
if someone wants resource with all the related resources in response then he has to marshall it by himself - IMHO if that's not the problem then fetching one resource with List instead of Fetch also shouldn't be.

Reply:
Ok, it looks like `Fetch` [uses list internally](https://github.com/cloudwan/gohan/blob/master/db/sql/sql.go#L1060), so that is fine,
but there is an inconsistency there with `LockList` which [sets join to true](https://github.com/cloudwan/gohan/blob/master/db/sql/sql.go#L948), when options are nil (in contrast to list).

Reply:
that code is pretty confusing because as [we build sql query few lines above](https://github.com/cloudwan/gohan/blob/master/db/sql/sql.go#L935) - here we are using it only to parse data. These lines don't make any sense as if we wanted related resources we should use join also in query. Let me just remove them